ibm/service/cos/resource_ibm_cos_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,708 @@
+		UpdateContext: resourceIBMCOSBucketLifecycleConfigurationUpdate,
+		DeleteContext: resourceIBMCOSBucketLifecycleConfigurationDelete,
+		Importer:      &schema.ResourceImporter{},
+		CustomizeDiff: customdiff.Sequence(
+			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+				return resourceValidateLifecycleRule(diff)
+			},
+		),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"bucket_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket CRN",
+			},
+			"bucket_location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket location",
+			},
+			"endpoint_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"public", "private", "direct"}),
+				Description:  "COS endpoint type: public, private, direct",
+				Default:      "public",
+			},
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"abort_incomplete_multipart_upload": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"days_after_initiation": {
+										Type:         schema.TypeInt,
+										ValidateFunc: validate.ValidateAllowedRangeInt(1, 3650),
+										Optional:     true,
+									},
+								},
+							},
+						},
+						"expiration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"date": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validate.ValidateUTCFormat,
+									},
+									"days": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validate.ValidateAllowedRangeInt(1, 3650),
+										Default:      0, // API returns 0
+									},
+									"expired_object_delete_marker": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"filter": {
+							Type:             schema.TypeList,
+							Required:         true,
+							DiffSuppressFunc: suppressMissingFilterConfigurationBlock,
+							// IBM has filter a required parameter
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"prefix": {
+										Type:     schema.TypeString, // check if prefix empty is eccepted and if filter empty is accepted
```

Comment:
is this comment needed? / check if prefix empty is eccepted and if filter empty is accepted